src/runaqa.ts

```diff
@@ -1,324 +1,475 @@
+ * @param  {[string]} buildList AQAvit Test suite
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @param  {[string]} openj9Repo Alternative openj9Repo
+ * @param  {[string]} tkgRepo Alternative TKG repo
+ * @param  {[string]} vendorTestParams Vendor provided test parameters
+ * @param  {[string]} aqasystemtestsRepo Alternative AQA-systemtestRepo
+ * @return {[null]}  null
+ */
+export async function setupParallelEnv(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string,
+  numMachines: string
+): Promise<void> {
+
+  await setupTestEnv(version, jdksource, customizedSdkUrl, sdkdir, buildList, aqatestsRepo, openj9Repo, tkgRepo, vendorTestParams, aqasystemtestsRepo);
+  process.chdir('TKG');
+  process.env.PARALLEL_OPTIONS = `PARALLEL_OPTIONS=TEST=${buildList} TEST_TIME= NUM_MACHINES=${numMachines}`;
+  await exec.exec(`make genParallelList ${process.env.PARALLEL_OPTIONS}`);
+
+}
+
+/**
+ * Sets up the test environment on the runner.
+ * @param  {string} version JDK Version being tested
+ * @param  {string} jdksource Source for JDK
+ * @param  {[string]} customizedSdkUrl Download link for JDK binaries
+ * @param  {[string]} sdkdir Directory for SDK
+ * @param  {[string]} buildList AQAvit Test suite
+ * @param  {[string]} aqatestsRepo Alternative aqatestRepo
+ * @param  {[string]} openj9Repo Alternative openj9Repo
+ * @param  {[string]} tkgRepo Alternative TKG repo
+ * @param  {[string]} vendorTestParams Vendor provided test parameters
+ * @param  {[string]} aqasystemtestsRepo Alternative AQA-systemtestRepo
+ * @return {null}  null
+ */
+async function setupTestEnv(
+  version: string,
+  jdksource: string,
+  customizedSdkUrl: string,
+  sdkdir: string,
+  buildList: string,
+  aqatestsRepo: string,
+  openj9Repo: string,
+  tkgRepo: string,
+  vendorTestParams: string,
+  aqasystemtestsRepo: string
+  ):  Promise<void>{
+    await installPlatformDependencies();
+    setSpec();
+    process.env.BUILD_LIST = buildList;
+    if ((jdksource === 'upstream' ||
+        jdksource === 'github-hosted' ||
+        jdksource === 'install-jdk') &&
+        !('TEST_JDK_HOME' in process.env)) {
+        process.env.TEST_JDK_HOME = getTestJdkHome(version, jdksource);
+    }
+    if (!('TEST_JDK_HOME' in process.env)) {
+        process.env.TEST_JDK_HOME = `${sdkdir}/openjdkbinary/j2sdk-image`;
+    }
+
+    await getAqaTestsRepo(aqatestsRepo, version, buildList);
+    await runGetSh(tkgRepo, openj9Repo, vendorTestParams, jdksource, customizedSdkUrl, sdkdir);
+    resetJDKHomeFromProperties();
+
+    //Get Dependencies, using /*zip*/dependents.zip to avoid loop every available files
+    let dependents = await tc.downloadTool('https://ci.adoptopenjdk.net/view/all/job/test.getDependency/lastSuccessfulBuild/artifact//*zip*/dependents.zip');

```

Comment:
should we parametrize the ci.adoptopenjdk.net URL so it can be overridden?

Reply:
which part of this could be parameterized? Might we have to add additional args to the GH action which calls this action? Or some additional variables in the function?

Reply:
I'm just wondering if you were to run this in house with no access to ci.adoptoipenjdk.net but an internal clone instead, does that help?  Probably something for a future PR though :-)

Reply:
I believe we put the behaviour to be if not able to access the public server, then download each of the dependencies individually.  Again, those would need access to public URLs.

Reply:
OK, we can resolve this later if required (we'll be trying in house over the coming weeks and can adjust)